acoustid/indexclient.py

```
@@ -87,20 +87,21 @@ def _connect(self):
-            self.sock.setblocking(0)
+            self.sock.setblocking(False)
         except socket.error as e:
             raise IndexClientError(
                 "unable to connect to the index server at %s:%s (%s)"
                 % (self.host, self.port, e)
             )
 
-    def _putline(self, line):
-        # type: (str) -> None
+    def _putline(self, line: str) -> None:
+        assert self.sock is not None
```

Ruff (0.8.2)

Use of `assert` detected

(S101)